src/go/reporter/reporter.go

```diff
@@ -0,0 +1,43 @@
+// Copyright 2021 The Magma Authors.
+//
+// This source code is licensed under the BSD-style license found in the
+// LICENSE file in the root directory of this source tree.
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package reporter
+
+import (
+	"time"
+)
+
+// EventID is a hexadecimal string representing a unique uuid4 for an Event.
+// An EventID must be 32 characters long, lowercase and not have any dashes.
+type EventID string
+
+// Level marks the severity of the event.
+type Level string
+
+// Breadcrumb specifies an application event that occurred before a reported event.
+// An event may contain one or more breadcrumbs.
+type Breadcrumb struct {
+	Type      string                 `json:"type,omitempty"`
+	Category  string                 `json:"category,omitempty"`
+	Message   string                 `json:"message,omitempty"`
+	Data      map[string]interface{} `json:"data,omitempty"`
+	Level     Level                  `json:"level,omitempty"`
+	Timestamp time.Time              `json:"timestamp"`
+}
+
+type Reporter interface {
```

Comment:
I wonder if we really need CaptureMessage and CaptureException

Btw, what do pkg/errors w/ stack look like when we send to sentry? if they get processed well, maybe we only have `Capture(err error)`?

Reply:
looks like it's really just for typing. https://github.com/getsentry/sentry-go/blob/8fbb027528f9464e0233afa756cef0f99bd70906/client.go#L323-L341
can probably get away with just a capture